internal/extproc/translator/openai_gcpanthropic.go

```
@@ -6,59 +6,692 @@
+			}
+		default:
+			err = fmt.Errorf("content type not supported: %v", content.Type)
+			return
+		}
+	}
+
+	// Handle tool_calls (if any).
+	for i := range openAiMessage.ToolCalls {
+		toolCall := &openAiMessage.ToolCalls[i]
+		var input map[string]interface{}
+		if err = json.Unmarshal([]byte(toolCall.Function.Arguments), &input); err != nil {
+			err = fmt.Errorf("failed to unmarshal tool call arguments: %w", err)
+			return
+		}
+		toolUse := anthropic.ToolUseBlockParam{
+			ID:    toolCall.ID,
+			Type:  "tool_use",
+			Name:  toolCall.Function.Name,
+			Input: input,
+		}
+		contentBlocks = append(contentBlocks, anthropic.ContentBlockParamUnion{OfToolUse: &toolUse})
+	}
+
+	return &anthropic.MessageParam{
+		Role:    anthropic.MessageParamRoleAssistant,
+		Content: contentBlocks,
+	}, nil
+}
+
+// openAIToAnthropicMessages converts OpenAI messages to Anthropic message params type, handling all roles and system/developer logic.
+func openAIToAnthropicMessages(openAIMsgs []openai.ChatCompletionMessageParamUnion) (anthropicMessages []anthropic.MessageParam, systemBlocks []anthropic.TextBlockParam, err error) {
+	for i := range openAIMsgs {
+		msg := openAIMsgs[i]
+		switch msg.Type {
+		case openai.ChatMessageRoleSystem:
+			if param, ok := msg.Value.(openai.ChatCompletionSystemMessageParam); ok {
+				devParam := systemMsgToDeveloperMsg(param)
+				systemBlocks = append(systemBlocks, anthropic.TextBlockParam{Text: extractSystemPromptFromDeveloperMsg(devParam)})
+			}
+		case openai.ChatMessageRoleDeveloper:
+			if param, ok := msg.Value.(openai.ChatCompletionDeveloperMessageParam); ok {
+				systemBlocks = append(systemBlocks, anthropic.TextBlockParam{Text: extractSystemPromptFromDeveloperMsg(param)})
+			}
+		case openai.ChatMessageRoleUser:
+			message := msg.Value.(openai.ChatCompletionUserMessageParam)
+			var content []anthropic.ContentBlockParamUnion
+			content, err = openAIToAnthropicContent(message.Content.Value)
+			if err != nil {
+				return
+			}
```

Are we suppressing the error? Can we do `return (nil, nil, err)`. Same for below.